### PR TITLE
add edit article endpoint

### PR DESCRIPTION
**Pull request by vorobevamv**
added edit article endpoint;
added EF Support

### CODE REVIEW

File: src/WikiService.Mappers/Models/PatchDbArticleMapper.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  public class PatchDbArticleMapper : IPatchDbArticleMapper
+  {
+    public JsonPatchDocument<DbArticle> Map(JsonPatchDocument<EditArticleRequest> request)
+    {
+      if (request == null)
+      {
+        return null;
+      }
+
+      var result = new JsonPatchDocument<DbArticle>();
```
Comment (MariyaSin) [inline src/WikiService.Mappers/Models/PatchDbArticleMapper.cs]: please, do not use var

Comment (vorobevamv) [reply src/WikiService.Mappers/Models/PatchDbArticleMapper.cs]: fixed

File: src/WikiService.Data/ArticleRepository.cs [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
 {
   public class ArticleRepository : IArticleRepository
   {
     private readonly IDataProvider _provider;
+    private readonly IHttpContextAccessor _httpContextAccessor;
 
     public ArticleRepository(
-      IDataProvider provider)
+      IDataProvider provider,
+       IHttpContextAccessor httpContextAccessor)
```
Comment (VLiamin) [inline src/WikiService.Data/ArticleRepository.cs]: space

File: src/WikiService.Data/ArticleRepository.cs [hunk 3]
```diff
@@ -37,9 +42,24 @@ public async Task<bool> DoesSameNameExistAsync(Guid rubricId, string articleName
...
     public async Task<DbArticle> GetAsync(Guid articleId)
-    {  
+    {
       return await _provider.Articles.Include(a => a.Files)
         .FirstOrDefaultAsync(article => article.Id == articleId);
     }
+
+    public async Task<bool> EditAsync(DbArticle dbArticle, JsonPatchDocument<DbArticle> request)
+    {
+      if (dbArticle == null || request == null)
```
Comment (VLiamin) [inline src/WikiService.Data/ArticleRepository.cs]: if (dbArticle is null || request is null)

File: src/WikiService.Mappers/Models/PatchDbArticleMapper.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    public JsonPatchDocument<DbArticle> Map(JsonPatchDocument<EditArticleRequest> request)
+    {
+      if (request == null)
+      {
+        return null;
+      }
+
+      JsonPatchDocument<DbArticle> result = new JsonPatchDocument<DbArticle>();
+
+      foreach (var item in request.Operations)
```
Comment (VLiamin) [inline src/WikiService.Mappers/Models/PatchDbArticleMapper.cs]: May be replace var?

File: src/WikiService.Mappers/Models/PatchDbArticleMapper.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.AspNetCore.JsonPatch.Operations;
+
+namespace LT.DigitalOffice.WikiService.Mappers.Models
+{
+  public class PatchDbArticleMapper : IPatchDbArticleMapper
+  {
+    public JsonPatchDocument<DbArticle> Map(JsonPatchDocument<EditArticleRequest> request)
+    {
+      if (request == null)
```
Comment (VLiamin) [inline src/WikiService.Mappers/Models/PatchDbArticleMapper.cs]: is null

File: src/WikiService.Validation/Article/EditArticleRequestValidator.cs [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+               _currentRubricId = item.path.EndsWith(nameof(EditArticleRequest.RubricId))
+                 ? Guid.TryParse(item.value?.ToString(), out Guid _rubricIdParseResult) ? _rubricIdParseResult : _currentRubricId
+                 : _currentRubricId;
+
+               _currentArticleName = item.path.EndsWith(nameof(EditArticleRequest.Name)) ? item.value?.ToString() : _currentArticleName;
+             }
+
+             return (_currentRubricId == x.Item1.RubricId && _currentArticleName == x.Item1.Name)
+             ? true
+             :!await _articleRepository.DoesSameNameExistAsync(_currentRubricId, _currentArticleName);
```
Comment (VLiamin) [inline src/WikiService.Validation/Article/EditArticleRequestValidator.cs]: space

File: src/WikiService.Validation/Article/EditArticleRequestValidator.cs [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.WikiService.Validation.Article
+{
+  public class EditArticleRequestValidator : ExtendedEditRequestValidator<DbArticle, EditArticleRequest>, IEditArticleRequestValidator
+  {
+    private readonly IArticleRepository _articleRepository;
+    private readonly IRubricRepository _rubricRepository;
+    private async Task HandleInternalPropertyValidationAsync(Operation<EditArticleRequest> requestedOperation, CustomContext context)
```
Comment (VLiamin) [inline src/WikiService.Validation/Article/EditArticleRequestValidator.cs:21]: empty line

**Comment by VLiamin** [comment]
May be change name of this PR to  Add edit article endpoint? Without feature?

File: src/WikiService.Data/ArticleRepository.cs (hunk 2 above)
Comment (vorobevamv) [reply src/WikiService.Data/ArticleRepository.cs]: fixed

File: src/WikiService.Mappers/Models/PatchDbArticleMapper.cs (hunk 4 above)
Comment (vorobevamv) [reply src/WikiService.Mappers/Models/PatchDbArticleMapper.cs]: done

File: src/WikiService.Data/ArticleRepository.cs (hunk 3 above)
Comment (vorobevamv) [reply src/WikiService.Data/ArticleRepository.cs]: done

File: src/WikiService.Mappers/Models/PatchDbArticleMapper.cs (hunk 5 above)
Comment (vorobevamv) [reply src/WikiService.Mappers/Models/PatchDbArticleMapper.cs]: done

File: src/WikiService.Validation/Article/EditArticleRequestValidator.cs (hunk 6 above)
Comment (vorobevamv) [reply src/WikiService.Validation/Article/EditArticleRequestValidator.cs]: fixed

File: src/WikiService.Validation/Article/EditArticleRequestValidator.cs (hunk 7 above)
Comment (vorobevamv) [reply src/WikiService.Validation/Article/EditArticleRequestValidator.cs:21]: fixed

**Review by VLiamin — APPROVED** [review]

File: src/WikiService.Models.Dto/Requests/Article/EditArticleRequest.cs [hunk 8]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace LT.DigitalOffice.WikiService.Models.Dto.Requests.Article
+{
+  public record EditArticleRequest
```
Comment (DarkFelas) [inline src/WikiService.Models.Dto/Requests/Article/EditArticleRequest.cs]: There is no need in record if u don't use its features :)
Whole this text can be simplified to:


```
using System;

namespace LT.DigitalOffice.WikiService.Models.Dto.Requests.Article;

public record EditArticleRequest(string Name, string Content, bool IsActive, Guid RubricId);
```

Comment (vorobevamv) [reply src/WikiService.Models.Dto/Requests/Article/EditArticleRequest.cs]: OK, fixed. thanks)

File: Common.Configurations.targets [hunk 9]
```diff
@@ -25,16 +25,10 @@
...
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="6.0.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.0" />
-  </ItemGroup>
-
   <ItemGroup Condition="$(AssemblyName.Contains('.UnitTests')) != 'true'">
-    <PackageReference Include="LT.DigitalOffice.Kernel" Version="1.1.60" />
-    <PackageReference Include="LT.DigitalOffice.Kernel.BrokerSupport" Version="1.0.5" />
+    <PackageReference Include="LT.DigitalOffice.Kernel" Version="1.1.61" />
+    <PackageReference Include="LT.DigitalOffice.Kernel.BrokerSupport" Version="1.0.6" />
+	<PackageReference Include="LT.DigitalOffice.Kernel.EFSupport" Version="1.0.1" />
```
Comment (MariyaSin) [inline Common.Configurations.targets]: fix tab

File: src/WikiService.Validation/Article/EditArticleRequestValidator.cs [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+
+      #region RubricId
+      await AddFailureForPropertyIfAsync(
+        nameof(EditArticleRequest.RubricId),
+        x => x == OperationType.Replace,
+        new()
+        {
+          {
+            async x => await _rubricRepository.DoesExistAsync(
+            Guid.TryParse(x.value?.ToString(), out Guid _rubricId) ? _rubricId : default
```
Comment (MariyaSin) [inline src/WikiService.Validation/Article/EditArticleRequestValidator.cs]: add tab

File: src/WikiService.Validation/Article/EditArticleRequestValidator.cs [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+                 ? Guid.TryParse(item.value?.ToString(), out Guid _rubricIdParseResult) ? _rubricIdParseResult : _currentRubricId
+                 : _currentRubricId;
+
+               _currentArticleName = item.path.EndsWith(nameof(EditArticleRequest.Name), StringComparison.OrdinalIgnoreCase)
+               ? item.value?.ToString()
+               : _currentArticleName;
+             }
+
+             return (_currentRubricId == x.Item1.RubricId && _currentArticleName == x.Item1.Name)
+             ? true
```
Comment (MariyaSin) [inline src/WikiService.Validation/Article/EditArticleRequestValidator.cs]: add tab

File: src/WikiService.Validation/Article/EditArticleRequestValidator.cs [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+             : !await _articleRepository.DoesSameNameExistAsync(_currentRubricId, _currentArticleName);
```
Comment (MariyaSin) [inline src/WikiService.Validation/Article/EditArticleRequestValidator.cs]: add tab

**Review by MariyaSin — APPROVED** [review]

**Review by MariyaSin — APPROVED** [review]

**Review by a5x55555 — APPROVED** [review]

**Review by AnnPostnikova — APPROVED** [review]

**Pull request merged**